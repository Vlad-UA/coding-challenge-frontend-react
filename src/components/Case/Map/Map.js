import React, { Component } from 'react';
import GoogleMapReact from 'google-map-react';
import PropTypes from 'prop-types';

// Instruments
import styles from './Map.module.scss';

export default class Map extends Component {
  static defaultProps = {
    center: {
      lat: 52.51444,
      lng: 13.3954199,
    },
  };

  render() {
    const { center } = this.props;

    return (
      <div className={styles.Map}>
        <GoogleMapReact
          bootstrapURLKeys={{ key: 'AIzaSyB23QiArAUU74ItBhbCBnc7gmBKXhdXumc' }}
          defaultCenter={center}
          defaultZoom={15}
        >
          <div className={styles.text}>* Incident location</div>
        </GoogleMapReact>
      </div>
    );
  }
}

// TODO add propTypes
Map.propTypes = {
  center: PropTypes.object.isRequired,
};
