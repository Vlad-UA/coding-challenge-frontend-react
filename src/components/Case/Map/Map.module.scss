.Map {
  height: 100%;
  width: 100%;

  .text {
    color: red;
    font-size: 14px;
    font-weight: bold;
    background-color: rgba(255, 0, 0, 0.1);
    padding: 4px;
    border-radius: 4px;
    display: inline-block;
  }

}
