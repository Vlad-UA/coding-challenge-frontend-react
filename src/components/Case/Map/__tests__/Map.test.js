// Core
import React from 'react';

// Test utilities
import renderer from 'react-test-renderer';

// Component
import Map from '../Map';

describe('Component Map:', () => {
  test('Must match snapshot:', () => {
    const map = renderer
      .create(<Map center={{ lat: 45.5164386, lng: -122.6244177 }} />)
      .toJSON();

    expect(map).toMatchSnapshot();
  });
});
