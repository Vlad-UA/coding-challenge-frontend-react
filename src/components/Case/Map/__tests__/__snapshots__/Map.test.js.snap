// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Map: Must match snapshot: 1`] = `
<div
  className="Map"
>
  <div
    onClick={[Function]}
    onMouseDownCapture={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "left": 0,
          "margin": 0,
          "padding": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
    <div
      style={
        Object {
          "height": "50%",
          "left": "50%",
          "margin": 0,
          "padding": 0,
          "position": "absolute",
          "top": "50%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "left": 0,
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <div
          $geoService={
            Geo {
              "hasSize_": false,
              "hasView_": true,
              "transform_": Transform {
                "_maxZoom": 52,
                "_minZoom": 0,
                "_zoom": 15,
                "angle": -0,
                "center": LatLng {
                  "lat": 45.5164386,
                  "lng": -122.6244177,
                },
                "height": 0,
                "latRange": Array [
                  -85.05113,
                  85.05113,
                ],
                "scale": 32768,
                "tileSize": 256,
                "tileZoom": 15,
                "width": 0,
                "zoomFraction": 0,
              },
            }
          }
          $onMouseAllow={[Function]}
          $prerender={true}
          className="text"
        >
          * Incident location
        </div>
      </div>
    </div>
  </div>
</div>
`;
